### PR TITLE
TS-4930: Unfolds request headers that are using obs continuations

### PR DESCRIPTION
**Pull request by zwoop**
I also removed a file that is basically a duplication of another file, and this duplicated file is not used at all. This confused me as I was testing and updating the regression tests. Basically, that old test app is already incorporated into the regression tests.

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/931/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/825/ for details.

File: proxy/hdrs/HdrTest.cc [hunk 1]
```diff
@@ -522,6 +524,16 @@ HdrTest::test_mime()
     return (failures_to_status("test_mime", 1));
   }
 
+  // Test the (new) continuation line folding to be correct. This should replace the
+  // \r\n with two spaces (so a total of three between "part1" and "part2").
+  int length;
+  const char *continuation = hdr.value_get("continuation", 12, &length);
+
+  if ((13 != length) || strncmp(continuation, "part1   part2", 13) || strncmp(continuation + 5, "   ", 3)) {
```
Comment (jpeach) [inline proxy/hdrs/HdrTest.cc]: This will be easier to debug if you separate each condition into a separate failure.

File: proxy/hdrs/HdrTest.cc [hunk 2]
```diff
@@ -474,7 +474,9 @@ HdrTest::test_url()
 int
 HdrTest::test_mime()
 {
-  static const char mime[] = {
+  // This can not be a static string (any more) since we unfold the headers
+  // in place.
```
Comment (jpeach) [inline proxy/hdrs/HdrTest.cc]: So the contents now get modified? If so, we need to pass in non-const pointers.

File: proxy/hdrs/MIME.cc [hunk 3]
```diff
@@ -2435,6 +2435,12 @@ mime_scanner_get(MIMEScanner *S, const char **raw_input_s, const char *raw_input
     case MIME_PARSE_AFTER:
       // After a LF. Might be the end or a continuation.
       if (ParseRules::is_ws(*raw_input_c)) {
+        char *unfold = const_cast<char *>(raw_input_c - 1);
+
+        *unfold-- = ' ';
+        if (ParseRules::is_cr(*unfold)) {
+          *unfold = ' ';
```
Comment (jpeach) [inline proxy/hdrs/MIME.cc]: We can't accept const data if we do this.

**Comment by zwoop** [comment]
Try build on freebsd again [approve ci].

**Comment by atsci** [comment]
FreeBSD build *failed*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/932/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/826/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/933/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/827/ for details.

**Comment by zwoop** [comment]
[approve ci]

**Pull request closed without merging**